merkle/merkle_test.go

```diff
@@ -0,0 +1,44 @@
+package merkle
+
+import (
+	"encoding/binary"
+	"testing"
+
+	"context"
+	"strconv"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/trace"
+	"github.com/ava-labs/avalanchego/x/merkledb"
+)
+
+var res_root ids.ID
+var res_db merkledb.MerkleDB
+var res_err error
+
+func BenchmarkMerkleTxRoot(b *testing.B) {
+	for _, size := range []int{10, 100, 1000, 10000} {
+    ctx := context.TODO()
+    tracer := trace.Noop
+    merkleItems := make([][]byte, 0, size)
+    for i := 0; i < size; i++ {
+      as_bytes := make([]byte, 32)
+      binary.BigEndian.PutUint32(as_bytes, uint32(i))
+      merkleItems = append(merkleItems, as_bytes)
+    }
+    var root ids.ID
+    var db merkledb.MerkleDB
+    var err error
+		b.Run(strconv.Itoa(size), func(b *testing.B) {
+			for n := 0; n < b.N; n++ {
```

Comment:
Can you also add `b.ReportAllocs()`

Reply:
done!

Reply:
For some reason I'm not able to see the memory allocation report with this activated on cd108423d752c9cf919ca964a29446bf8c73711c

Reply:
That only works when passing the `-test.benchmem` flag